### PR TITLE
refactor: replace react-calendar date picker with MUI date picker com…

### PR DESCRIPTION
**Pull request by Ngoakor12**
…ponent

Pull-Request for `paretOS`

## Description
- Replaced react-calendar with MUI date picker component
- Added packages: @mui/x-date-pickers, @date-io/date-fns, date-fns

## Relates to
- ##200

## Reviewers
- @mikhael28 
- @jayeclark 

## Screenshots / other info
image

That datepicker component is the closest one I found that resembles the previous calendar. So I wanted to know if it's alright and perhaps why a calendar instead of a simple date input component?

There are also a few react-bootstrap components, I want to know if I should replace them as well or if they're already in other issues or someone can open a separate issue on replacing them.

There also doesn't seem to be any validation, choosing a date simply enables the create button even though the first two fields are not filled.

### CODE REVIEW

**Comment by mikhael28** [comment]
Hey @Ngoakor12 thank you for sending this in - sorry for the delay in response, I've been traveling and shipping code at my day job. I'm going to pull this down tomorrow and review it - my only concern would be the bundle-size implication of the Datepicker from Material UI vs the Calendar that I already have. That, and the ease of adding validation logic - theoretically, not every 'template' should be able to be started on any date - some MUST start on Monday, Friday, etc. Will review tomorrow!

File: package.json [hunk 1]
```diff
@@ -19,6 +21,7 @@
     "@testing-library/jest-dom": "^5.16.2",
     "@testing-library/react": "^12.1.4",
     "classnames": "2.3.1",
+    "date-fns": "^2.28.0",
```
Comment (mikhael28) [inline package.json:24]: Is this package necessary? I'm not seeing it imported anywhere - is it a dependency of the mui/x date picker?

File: package.json [hunk 2]
```diff
@@ -7,10 +7,12 @@
     "@aws-amplify/auth": "^4.4.2",
     "@aws-amplify/core": "^4.4.0",
     "@aws-amplify/storage": "^4.4.16",
+    "@date-io/date-fns": "^2.13.1",
```
Comment (mikhael28) [inline package.json:10]: Same here

File: src/arena/SprintCreation.js [hunk 3]
```diff
@@ -327,19 +328,21 @@ function SprintCreation(props) {
...
-        }}
-        value={startDate}
-        maxDetail="month"
-        minDetail="month"
-        // minDate={new Date()}
-        maxDate={new Date(Date.now() + 2592000000)}
-        // tileDisabled={({ date }) => date.getDay() !== 1}
-        showNeighboringMonth
-      />
+      <LocalizationProvider dateAdapter={AdapterDateFns}>
```
Comment (mikhael28) [inline src/arena/SprintCreation.js:331]: This is great! In terms of bundle size, it is around the same the `react-calendar` was, around 50kb but the theming should be much easier to fix.

File: package.json [hunk 4]
```diff
@@ -19,6 +21,7 @@
     "@testing-library/jest-dom": "^5.16.2",
     "@testing-library/react": "^12.1.4",
     "classnames": "2.3.1",
+    "date-fns": "^2.28.0",
     "flutterwave-react-v3": "1.2.0",
     "history": "5.3.0",
     "lodash.clonedeep": "4.5.0",
```
Comment (mikhael28) [inline package.json:27]: Can you remove the `react-calendar` npm package too, and re-run yarn install to re-generate yarn.lock? Then push up the changes.

File: src/arena/SprintCreation.js (hunk 3 above)
Comment (mikhael28) [reply src/arena/SprintCreation.js:331]: Can confirm the functionality seems to be working, great work!

**Review by mikhael28 — CHANGES_REQUESTED** [review]
Can you please update the package.json based on my comments, and answer the questions I have about `date-fns`?

File: package.json (hunk 1 above)
Comment (Ngoakor12) [reply package.json:25]: Yes, all the date related packages added are dependencies of @mui/x date picker.

**Review by mikhael28 — APPROVED** [review]
Great, I've gone ahead and approved an am merging! Great work, thank you for answering my questions.

**Pull request merged**